stdlib/public/SwiftShims/LibcShims.h

```diff
@@ -49,7 +49,9 @@ typedef      long int __swift_ssize_t;
 
 // This declaration might not be universally correct.
 // We verify its correctness for the current platform in the runtime code.
-#if defined(__linux__)
+#if defined(__ANDROID__)
+typedef __swift_uint16_t __swift_mode_t;
+#elif defined(__linux__)
```

Comment:
This feels a bit fragile to me.  This may seem stylistic and silly, but, I feel that if someone were to accidentally reorder the items, this would break.  Writing it as the following would make that more visible:

```
  #if defined(__linux__)
  # if defined(__ANDROID__)
  typedef __swift_uint16_t __swift_mode_t;
  # else
  typedef __swift_uint32_t __swift_mode_t;
  # endif
  #else
```

Reply:
absolutely, doing that change now! I'll keep this mind for the next changes.

Reply:
@compnerd got it, done. I'll put a note to do a cleanup PR later fixes all the other occurrences (or as much as possible w/o breaking anything :) )